tsdb/shard_test.go

```diff
@@ -1628,64 +1648,168 @@ func TestMeasurementFieldSet_Corrupt(t *testing.T) {
+		defer checkMeasurementFieldSetClose(t, mf)
+		measurement := []byte("cpu")
+		fields := mf.CreateFieldsIfNotExists(measurement)
+		fieldName := "value"
+		if err := fields.CreateFieldIfNotExists([]byte(fieldName), influxql.Float); err != nil {
 			t.Fatalf("create field error: %v", err)
 		}
+		change := tsdb.FieldChange{
+			FieldCreate: tsdb.FieldCreate{
+				Measurement: []byte(measurement),
+				Field:       &tsdb.Field{ID: 0, Name: fieldName, Type: influxql.Float},
+			},
+			ChangeType: tsdb.AddMeasurementField,
+		}
 
-		if err := mf.Save(); err != nil {
+		if err := mf.Save(tsdb.FieldChanges{&change}); err != nil {
 			t.Fatalf("save error: %v", err)
 		}
 	}()
 	stat, err := os.Stat(path)
 	if err != nil {
 		t.Fatalf("stat error: %v", err)
 	}
-	// Truncate the file to simulate a a corrupted file
+	// Truncate the file to simulate a corrupted file
 	if err := os.Truncate(path, stat.Size()-3); err != nil {
 		t.Fatalf("truncate error: %v", err)
 	}
-	mf, err := tsdb.NewMeasurementFieldSet(path)
+	mf, err := tsdb.NewMeasurementFieldSet(path, nil)
 	if err == nil {
 		t.Fatal("NewMeasurementFieldSet expected error")
 	}
-	defer mf.Close()
+	defer checkMeasurementFieldSetClose(t, mf)
 
 	fields := mf.FieldsByString("cpu")
 	if fields != nil {
 		t.Fatal("expecte fields to be nil")
 	}
 }
+
+func TestMeasurementFieldSet_CorruptChangeFile(t *testing.T) {
+	dir, cleanup := MustTempDir()
+	defer cleanup()
+
+	testFields := []struct {
+		Measurement string
+		Field       string
+		FieldType   influxql.DataType
+	}{
+		{
+			Measurement: "cpu",
+			Field:       "value_1",
+			FieldType:   influxql.Float,
+		},
+		{
+			Measurement: "cpu",
+			Field:       "value_2",
+			FieldType:   influxql.String,
+		},
+		{
+			Measurement: "cpu",
+			Field:       "value_3",
+			FieldType:   influxql.Integer,
+		},
+	}
+
+	path := filepath.Join(dir, "fields.idx")
+	var mf *tsdb.MeasurementFieldSet
+	var err error
+	mf, err = tsdb.NewMeasurementFieldSet(path, nil)
+	if err != nil {
+		t.Fatalf("NewMeasurementFieldSet error: %v", err)
+	}
+	defer checkMeasurementFieldSetClose(t, mf)
+	for _, f := range testFields {
+		fields := mf.CreateFieldsIfNotExists([]byte(f.Measurement))
+		if err := fields.CreateFieldIfNotExists([]byte(f.Field), f.FieldType); err != nil {
+			t.Fatalf("create field error: %v", err)
+		}
+		change := tsdb.FieldChange{
+			FieldCreate: tsdb.FieldCreate{
+				Measurement: []byte(f.Measurement),
+				Field:       &tsdb.Field{ID: 0, Name: f.Field, Type: f.FieldType},
+			},
+			ChangeType: tsdb.AddMeasurementField,
+		}
+
+		if err := mf.Save(tsdb.FieldChanges{&change}); err != nil {
+			t.Fatalf("save error: %v", err)
+		}
+	}
+	changeFile := filepath.Join(dir, tsdb.FieldsChangeFile)
+	stat, err := os.Stat(changeFile)
+	if err != nil {
+		t.Fatalf("stat error: %v", err)
+	}
+	// Truncate the file to simulate a corrupted file
+	if err := os.Truncate(changeFile, stat.Size()-3); err != nil {
```

Comment:
FWIW I find it more ergonomic to write

```
require.NoError(t, os.Truncate(changeFile, stat.Size()-3))
```

No need to change it, but do you prefer the `if` style with `Fatalf` or are you just following the style for the rest of the file?

Reply:
I prefer the assert, but only started using them in the last few changes.  So, keeping the old style until I got fed up with it, then moving to assert.